### DISCUSSION TITLE
Making multi page/directory template?

### DISCUSSION THREAD

**Discussion by b1mind** [SubZtep/vite-plugin-pug#2] [Ideas]
A good way to include a baseUrl and keep relative file pathing in vite and use with a multi page template using pug? I have added more entry points to vite to point to the other .html files but having a hard time getting it working right, specially if you wanted sub directories.

Any ideas on this?

**Comment by SubZtep** [comment]
The [renderer](https://pugjs.org/api/reference.html#pugcompilesource-options) takes a data object argument, if I understand well these are _key/value_ pairs of local variables in Pug, exactly what you are looking for! I'm implementing it and will publish a new version soon**ish**.

Vite hardcoded `.html` extension for HTML, but creating Pug templates and using them like a higher-order component could be possible. It's a great idea, I will have a look!

**Reply by SubZtep** [reply]
@b1m1nd I made a new [release](https://www.npmjs.com/package/vite-plugin-pug/v/0.2.0), minimum viable update for now. Not sure what format would be the best for templates yet.

Set variables:
https://github.com/SubZtep/vite-plugin-pug/blob/bcef4127e002cb98d155dbd6abfd08a688234a18/examples/multipage/vite.config.ts#L6

Use them in a `.pug` file:
https://github.com/SubZtep/vite-plugin-pug/blob/bcef4127e002cb98d155dbd6abfd08a688234a18/examples/multipage/index.pug#L4-L6

**Reply by b1mind** [reply]
This should help a lot, I had just been making a templates folder at root and using import. extends. I will play around and see if I have any worth while feedback for ya.

**Reply by SubZtep** [reply]
@b1m1nd here is a snippet for templates but it doesn’t tickle my fancy https://github.com/SubZtep/vite-plugin-pug/tree/multipage/examples/multiroot

**Reply by b1mind** [reply]
This looks to be much better than how I was doing it (with multi-entry points and having to tell vite to look at them). That router looks interesting, seems like vite will not do things like snowpack in terms of html templating. I am still pretty new to JS and tooling still so, lots goes above my head.

Will this allow for import/extending from other pug files still? The issue I am having atm is if I want to extend the html head to include a different .js or link diff stylesheet it has to be a html entry point for vite to compile it during build time.

Here is my repo I am playing with, I got it partially working. Look forward to trying this new branch of yours though. (mind the mess I was using it real fast for a codepen challenge lol)
https://github.com/b1m1nd/vite-pug

**Reply by SubZtep** [reply]
I think introducing the router was an unnecessary step, it caused unwanted complexity for this script. At this level, there are much [better solutions](https://github.com/antfu/vitesse).

**Comment by SubZtep** [comment]
Hey, check out the new release (v0.3), [this pr](https://github.com/SubZtep/vite-plugin-pug/pull/4) resolve this issue. :)
Surely post any related comment/idea here.
